src/astro/databases/aws/redshift.py

```diff
@@ -0,0 +1,115 @@
+"""AWS Redshift table implementation."""
+from typing import List
+
+import pandas as pd
+import sqlalchemy
+from airflow.providers.amazon.aws.hooks.redshift_sql import RedshiftSQLHook
+from sqlalchemy import create_engine
+from sqlalchemy.engine.base import Engine
+
+from astro.constants import DEFAULT_CHUNK_SIZE, LoadExistStrategy
+from astro.databases.base import BaseDatabase
+from astro.files import File
+from astro.settings import REDSHIFT_SCHEMA
+from astro.sql.table import Metadata, Table
+
+DEFAULT_CONN_ID = RedshiftSQLHook.default_conn_name
+
+
+class RedshiftDatabase(BaseDatabase):
+    """
+    Handle interactions with Redshift databases.
+    """
+
+    DEFAULT_SCHEMA = REDSHIFT_SCHEMA
+
+    illegal_column_name_chars: List[str] = ["."]
+    illegal_column_name_chars_replacement: List[str] = ["_"]
+
+    def __init__(self, conn_id: str = DEFAULT_CONN_ID):
+        super().__init__(conn_id)
+        self._create_table_statement: str = "CREATE TABLE {} AS {}"
+
+    @property
+    def sql_type(self):
+        return "redshift"
+
+    @property
+    def hook(self) -> RedshiftSQLHook:
+        """Retrieve Airflow hook to interface with the Redshift database."""
+        return RedshiftSQLHook(redshift_conn_id=self.conn_id, use_legacy_sql=False)
+
+    @property
+    def sqlalchemy_engine(self) -> Engine:
+        """Return SQAlchemy engine."""
+        uri = self.hook.get_uri()
+        return create_engine(uri)
+
+    @property
+    def default_metadata(self) -> Metadata:
+        """Fill in default metadata values for table objects addressing redshift databases"""
+        # TODO: Change airflow RedshiftSQLHook to fetch database and schema separately.
```

Comment:
We should comment the limitation here from airflow RedshiftSQLHook side that it treats both schema and database in the same way.